test/Database/LSMTree/Model/Session.hs

```diff
@@ -280,6 +284,7 @@ instance Eq Err where
     where
       _coveredAllCases x = case x of
           ErrTableClosed{}          -> ()
+          ErrSnapshotCorrupted{}    -> ()
```

Comment:
You didn't add a case `(==) ErrSnapshotCorrupted ErrSnapshotCorrupted`

Reply:
Woops. Fixed.

Reply:
What's the deal with `_coveredAllCases`?

Reply:
The `Eq` instance is custom so if `Err` changes, the compiler doesn't always tell you to modify the `Eq` instance. With `_coveredAllCases`, the code won't compile when you've added or changed constructors

Reply:
So it's just a of future-proofing of the code